xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/template/InternalTemplateManager.java

```diff
@@ -894,4 +895,20 @@ public Template getTemplate(String templateName)
 
         return template;
     }
+
+    /**
+     * Create a DefaultTemplate with the given content and the given author.
+     *
+     * @param content the template content
+     * @param author the template author
+     * @return the template
+     */
+    public Template createStringTemplate(String content, DocumentReference author)
+    {
+        DefaultTemplate newTemplate = new DefaultTemplate(null);
```

Comment:
Ok, thanks for your explanation @tmortagne.

So, I created a private class in the `InternalTemplateManager` : 
```
private class StringTemplate extends DefaultTemplate
{
    StringTemplate(String content) {
        super(new StringResource(content));
    }
}
```
The `StringResource` class is very basic, I pasted the implementation here : https://gist.github.com/aubincleme/cbafc6f2f2bdeca8fef227cb1f847ccf.

I think that I’m now correctly instanciating a `Template`, as a matter of fact, I can create my `StringTemplate` and assign it the correct author with two instructions : 
```
DefaultTemplate newTemplate = new StringTemplate(content);
newTemplate.content.setAuthorReference(author);
```

Unfortunately, I also found out that adding the new method `createStringTemplate` to the `TemplateManager` interface would make RevAPI scream during maven build (I forgot that `TemplateManager` was a public API). 

So, do we have a particular workflow in this case ? I’m not sure that adding the method directly to `DefaultTemplateManager` is the best idea, but I also dont’t think that it would be nice to directly access `InternalTemplateManager#createStringTemplate` directly in my extension code, as it relies a lot on the old core.